ykrt/src/mt.rs

```diff
@@ -599,16 +624,49 @@ impl MT {
                         HotLocationKind::Tracing => {
                             let hl = loc.hot_location_arc_clone().unwrap();
                             match &*mtt.tstate.borrow() {
-                                MTThreadState::Tracing { hl: thread_hl, .. } => {
+                                MTThreadState::Tracing {
+                                    hl: thread_hl,
+                                    frameaddr: tracing_frameaddr,
+                                    ..
+                                } => {
                                     // This thread is tracing something...
                                     if !Arc::ptr_eq(thread_hl, &hl) {
                                         // ...but not this Location.
                                         TransitionControlPoint::NoAction
                                     } else {
-                                        // ...and it's this location: we have therefore finished
-                                        // tracing the loop.
-                                        lk.kind = HotLocationKind::Compiling;
-                                        TransitionControlPoint::StopTracing
+                                        // ...and it's this location...
+                                        match STACK_DIRECTION {
+                                            StackDirection::GrowsToHigherAddress => todo!(),
+                                            StackDirection::GrowsToLowerAddress => {
+                                                if frameaddr <= *tracing_frameaddr {
```

Comment:
So it's OK to stop tracing in deeper recursive calls?

I'm not sure. Doesn't that mean the trace looping will loop across frames, but without including the "bubbling up", which may be important for program sematics?

e.g.

```
f() {
  loop {
   control_point()
    if somthing {
      f();
    }
  }
  important()
}
```

So here, if you start tracing at frame depth 0, and stop and frame depth (say) 10, does it matter that the trace doesn't include `important()`? I'm not sure.

Reply:
As long as the trace looper is happy with this, I am.